### PR TITLE
A handful of tiny fixes.

### PR DESCRIPTION
**Pull request by MasterQ32**
`DynamicQuery` didn't handle all functions properly

### CODE REVIEW

File: sqlite.zig [hunk 1]
```diff
@@ -1452,8 +1452,8 @@ pub const DynamicStatement = struct {
     ///
     /// Possible errors:
     /// - SQLiteError.SQLiteNotFound if some fields not found
-    pub fn iterator(self: *Self, comptime Type: type, values: anytype) !Iterator(Type) {
-        try self.smartBind(values);
+    pub fn iterator(self: *Self, comptime Type: type, options: anytype, values: anytype) !Iterator(Type) {
```
Comment (vrischmann) [inline sqlite.zig:1455]: I would prefer we add `iteratorAlloc`  like [here](https://github.com/vrischmann/zig-sqlite/blob/master/sqlite.zig#L1724-L1733). We stopped exposing `options: anytype` to the user and instead clearly separate the allocation-capable function from the other.

Comment (MasterQ32) [reply sqlite.zig:1455]: Sure, i'll change it 🙂

Comment (MasterQ32) [reply sqlite.zig]: Added a second `iteratorAlloc` function

**Comment by vrischmann** [comment]
Thanks !

**Pull request merged**